### ISSUE TITLE
Incompatible with Python 2.7

### ISSUE THREAD

**Issue by chriscrowe** [rapidfuzz/RapidFuzz#40]
Can't install the package seemingly because of the encoding parameter being supplied to `open()` in the setup.py file:

```
    Traceback (most recent call last):
      File "<string>", line 17, in <module>
      File "build/rapidfuzz/setup.py", line 7, in <module>
        with open(path.join(this_dir, "VERSION"), encoding='utf-8') as version_file:
    TypeError: 'encoding' is an invalid keyword argument for this function
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

  File "<string>", line 17, in <module>

  File "/build/rapidfuzz/setup.py", line 7, in <module>

    with open(path.join(this_dir, "VERSION"), encoding='utf-8') as version_file:

TypeError: 'encoding' is an invalid keyword argument for this function
```

I also tried to install from source but it seems I don't have the necessary levenshtein.hpp header files

**Comment by maxbachmann** [comment]
Yes the following changes are required to add python 2.7 support:
- [x] as you noted the setup.py has to use `from io import open`
- [x] process.py is not allowed to used typing since this was introduced in python3.5
- [x] The C++ components make heavy use of the python c api for performance reasons. However since there are a lot of api differences between the way strings are implemented in python2 and python3 this api changed a lot aswell. -> There has to be a implementation of these parts both for python3 and python2

I did not do this yet, since python2 is end of life and I personally did not need it. However since now there is need for it, I did just add support for python 2.7 in version 0.11.0 :tada:  (wheels only exist for Linux and MacOS when using python 2.7 so far).
